rfcs/0000-standardize_the_kuadrant_spec.md

```diff
@@ -0,0 +1,238 @@
+      enable: []
+      disable: []
+    - level: "debug"
+      configure: true
+      enable: []
+      disable: []
+```
+The list based method seems to have more flexibility than the field based method.
+As the level is defined as a string and not a key, if there is a new component with different log levels added, updating to support those levels does not require updating the CRD.
+However, the first approach is more user readable. 
+
+One thing that would be left up to the feature design, and is out of scope of this RFC is how the elements should be prioritized. 
+For example if a component is listed as enable in two sets of configures, which configuration is added.
+In the case of logging I believe it should be the more verbose configuration.
+```yaml
+spec:
+  logging_level:
+    - level: "error"
+      configure: true
+      enable: ["kuadrant-operator"]
+      disable: []
+    - level: "debug"
+      configure: true
+      enable: ["kuadrant-operator"]
+      disable: []
+```
+
+## Other features
+The previous examples give a good example of how the API design is layout, and can be extended. 
+However, there are a number of other features that could be supported by this API design.
+These are mention here, but their structure is not shown.
+
+- Multiply gateway support and configuration. [https://github.com/Kuadrant/architecture/pull/7](https://github.com/Kuadrant/architecture/pull/7)
+- Custom installation of Kuadrant. [https://github.com/Kuadrant/kuadrant-operator/issues/71](https://github.com/Kuadrant/kuadrant-operator/issues/71)
+- Installation of dashboards. [https://github.com/Kuadrant/kuadrant-operator/issues/1067](https://github.com/Kuadrant/kuadrant-operator/issues/1067)
+- Installation of alerts.
+- Tracing [https://github.com/Kuadrant/architecture/pull/96](https://github.com/Kuadrant/architecture/pull/96)
+- High availability [https://github.com/Kuadrant/kuadrant-operator/issues/798](https://github.com/Kuadrant/kuadrant-operator/issues/798)
+- Observability [https://github.com/Kuadrant/architecture/pull/97](https://github.com/Kuadrant/architecture/pull/97)
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The API design will not cover every use that is needed, and once the case is hit the UX of the API will become harder.
+If this design is not followed and every feature defines its own design the UX for the user will always be hard, that in itself makes it normal.
+
+This also reduces the level of design needed to add new features to the kuadrant CR, which could lead to over adding of features.
+The over adding of features brings a maintenances cost on the operator that we may not be willing to support.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+
+# Prior art
+[prior-art]: #prior-art
```

Comment:
I took inspiration for the grafana config from the kiali CR https://kiali.io/docs/configuration/kialis.kiali.io/
I've used the prometheus and alertmanager CRs a lot https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api-reference/api.md#monitoringcoreoscomv1
Both seem like reasonable prior art in this space.

Reply:
I can add them is as prior art. 

The kiali CR reference shows how out of control this can get. That is only for one product while we would need to be able to do that for many.

Reply:
> how out of control this can get

What makes you say it's out of control?
From the [developer docs](https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-operator-resource), it looks like they have a good handle on managing configuration.
Perhaps it's different from those on the ground developing it, but from a user point of view, the docs have the entire spec [in one place](https://kiali.io/docs/configuration/kialis.kiali.io/), and detailed docs for the different sections, for example, https://kiali.io/docs/configuration/p8s-jaeger-grafana/grafana/

> That is only for one product while we would need to be able to do that for many.

Do you mean one component?
Kuadrant is one project, with multiple components in my view.